### PR TITLE
feat: add support to detect unused CRDs

### PR DESCRIPTION
**Pull request by m-yosefpor**
fixes https://github.com/yonahd/kor/issues/103

### CODE REVIEW

**Comment by yonahd** [comment]
I reviewed the code on a high level and it looks overall good.(we'll do a proper code review in the next few days hopefully)

The question here is whether presenting the user with individual unused CRDs adds value.

Take `monitoring.coreos.com` as an example, under it there are multiple CRDs e.g.
- Prometheus
- PrometheusRule
- ServiceMonitor

If only one of the above is not used can the user uninstall that individual CRD or is he bound by the higher level of `monitoring.coreos.com`?

Maybe we should be returning only the higher-level that all it's CRDs are not used, WDYT?

**Comment by m-yosefpor** [comment]
> I reviewed the code on a high level and it looks overall good.(we'll do a proper code review in the next few days hopefully)
> 
> The question here is whether presenting the user with individual unused CRDs adds value.
> 
> Take `monitoring.coreos.com` as an example, under it there are multiple CRDs e.g.
> 
>     * Prometheus
> 
>     * PrometheusRule
> 
>     * ServiceMonitor
> 
> 
> If only one of the above is not used can the user uninstall that individual CRD or is he bound by the higher level of `monitoring.coreos.com`?
> 
> Maybe we should be returning only the higher-level that all it's CRDs are not used, WDYT?

It actually really depends.. for example for the monitoring.coreos.com, the prometheus operator neeeds all the CRDs to exist.. but for other project (e.g. argoproj.io) it has many components (argo-workflow, argo-events, argo-cd, argo-appset controller), which all of those CRDs are in the same higher level api group, but they can be deployed independently and each of those use their own set of CRDs in that api group. So I think to cover all such cases, it's better to left the decision on whether he can remove the CRDs or not, to user and we will only report on a per CRD basis.

File: pkg/kor/kor.go [hunk 1]
```diff
@@ -80,6 +82,78 @@ func GetKubeClient(kubeconfig string) *kubernetes.Clientset {
 	return clientset
 }
 
+func GetAPIExtensionsClient(kubeconfig string) *apiextensionsclientset.Clientset {
```
Comment (yonahd) [inline pkg/kor/kor.go:85]: This is very similar code to `GetKubeClient` function
Let's create a function that gets the config and we can create different clientsets based off this function

Same for dynamic client

**Review by yonahd — COMMENTED** [review]
Looks good overall.
Please rebase and see the comment for making this code cleaner

**Comment by yonahd** [comment]
Also if you can add a couple test for this that would be great!

**Comment by yonahd** [comment]
@m-yosefpor Do you intend to finish this MR?

**Pull request closed without merging**